core/sql/executor/ExExeUtilGet.cpp

```diff
@@ -3074,18 +3074,21 @@ short ExExeUtilGetHbaseObjectsTcb::work()
 
         case PROCESS_NEXT_ROW_:
           {
-            if (currIndex_ == bal_->getSize())
+            if (currIndex_ == hbaseTables_->entries())
               {
                 step_ = DONE_;
                 break;
               }
 
-            Int32 len = 0;
-            hbaseName_ = bal_->getEntry(currIndex_, hbaseNameBuf_, 
-                                        ComMAX_3_PART_EXTERNAL_UTF8_NAME_LEN_IN_BYTES+6, 
-                                        len);
-            hbaseName_[len] = 0;
-            
+            Int32 len = hbaseTables_->at(currIndex_).len;
+            char *tmp  = hbaseTables_->at(currIndex_).val;
+            len = hbaseTables_->at(currIndex_).len;
```

Comment:
I hope the C++ compiler does common subexpression elimination so we don't keep re-evaluating the "at".